src/corefx/System.Globalization.Native/pal_calendarData.c

```diff
@@ -447,32 +432,25 @@ bool EnumAbbrevEraNames(const char* locale,
     while (true)
     {
         UErrorCode status = U_ZERO_ERROR;
+        char* name = GetCalendarName(calendarId);
 
-        UResourceBundle* rootResBundle = ures_open(nullptr, localeNamePtr, &status);
-        UResourceBundleHolder rootResBundleHolder(rootResBundle, status);
-
-        UResourceBundle* calResBundle = ures_getByKey(rootResBundle, "calendar", nullptr, &status);
-        UResourceBundleHolder calResBundleHolder(calResBundle, status);
-
-        UResourceBundle* targetCalResBundle =
-            ures_getByKey(calResBundle, GetCalendarName(calendarId), nullptr, &status);
-        UResourceBundleHolder targetCalResBundleHolder(targetCalResBundle, status);
-
-        UResourceBundle* erasColResBundle = ures_getByKey(targetCalResBundle, "eras", nullptr, &status);
-        UResourceBundleHolder erasColResBundleHolder(erasColResBundle, status);
-
-        UResourceBundle* erasResBundle = ures_getByKey(erasColResBundle, "narrow", nullptr, &status);
-        UResourceBundleHolder erasResBundleHolder(erasResBundle, status);
+        UResourceBundle* rootResBundle = ures_open(NULL, localeNamePtr, &status);
+        UResourceBundle* calResBundle = ures_getByKey(rootResBundle, "calendar", NULL, &status);
+        UResourceBundle* targetCalResBundle = ures_getByKey(calResBundle, name, NULL, &status);
+        UResourceBundle* erasColResBundle = ures_getByKey(targetCalResBundle, "eras", NULL, &status);
+        UResourceBundle* erasResBundle = ures_getByKey(erasColResBundle, "narrow", NULL, &status);
```

Comment:
Does this logic work fine? I mean the status values would be accumulated and note overridden? for example if ures_open failed and then some how the other subsequent calls succeed for any reason which will mark status as success. 

would it be safer to check the status after every call? and don't have to continue of one failed?

Reply:
This logic works and it is actually the suggested error handling pattern if you look at ICU samples. We can use it in more places if we wanted it to make the code smaller and a bit faster. I agree that it looks like a suspect outlier now since nothing else in the file is using the ICU error handling pattern. There should be either comment, or it should use the same error handling pattern like the rest of the file.

Reply:
> We can use it in more places if we wanted it to make the code smaller and a bit faster

Yes @jkotas, that was your comment and I cleaned it up from all places. Do you have any other place in mind which can be cleaned up?

Reply:
Yes, I have commented about the ICU style error handling, but there was not any cleanup done to use it. The only place that uses the ICU style error handling in the file right now is this one.

You can read more about the design of the ICU error handling here: http://userguide.icu-project.org/design#TOC-Error-Handling. Here is one example from many where the ICU style error handling can be used to make the code smaller and a bit faster:

```
int32_t GlobalizationNative_GetLatestJapaneseEra()
{
    UErrorCode err = U_ZERO_ERROR;
    UCalendar* pCal = ucal_open(NULL, 0, JAPANESE_LOCALE_AND_CALENDAR, UCAL_TRADITIONAL, &err);
    int32_t ret = ucal_getLimit(pCal, UCAL_ERA, UCAL_MAXIMUM, &err);
    ucal_close(pCal);
    return U_SUCCESS(err) ? ret : 0;
}
```

Reply:
> there was not any cleanup done to use it.

@jkotas, are you sure about that or am I missing something: https://github.com/dotnet/coreclr/pull/18614/commits/a8ebe29b983d793b4d0d27e31000d1afdce5923b?

Reply:
That cleanup just removed unreachable code - to match what the C++ holders were doing before. It did switch the code to use the ICU-style error handling.

Reply:
do you want to address it as part of this PR? I thought this pull request is approved..

Reply:
I think this needs one of the following:
- Change the error handling style here to match the style used in the rest of the change
- Or change the error handling style throughout the change to use ICU-style

Reply:
>  I thought this pull request is approved.

I have approved it because of I did not have any more comments. However, I have looked at this change multiple times and I knew that I am likely missing things (it is "reviewers faitigue") since this change had many bugs and comments already. It is why I have asked more folks for review.

Reply:
What I meant is there are many comments in this PR thread. It is hard for me to follow as well. I can do a follow up PR to address the rest of comments from green point marker onwards.

---

> Here is one example from many where the ICU style error handling can be used to make the code smaller and a bit faster:

is it about removing:

```c
if (U_FAILURE(err))
    return 0;
```

after `*_open` APIs calls?

Reply:
Yes, the `if (U_FAILURE(err))` checks are not needed in some cases after `*_open` calls thanks to  the ICU error handling model. http://userguide.icu-project.org/design#TOC-Error-Handling.

> I can do a follow up PR to address the rest of comments from green point marker onwards.

sounds good.